# COVID Project 

- This is a data exploration completed in SQL. Database taken from the 'Our world in data'. 
https://ourworldindata.org/covid-deaths

## Data exploration

In [5]:
SELECT *
FROM covid_deaths.csv

iso_code continent  ... life_expectancy human_development_index
0          AFG      Asia  ...           64.83                   0.511
1          AFG      Asia  ...           64.83                   0.511
2          AFG      Asia  ...           64.83                   0.511
3          AFG      Asia  ...           64.83                   0.511
4          AFG      Asia  ...           64.83                   0.511
...        ...       ...  ...             ...                     ...
85166      ZWE    Africa  ...           61.49                   0.571
85167      ZWE    Africa  ...           61.49                   0.571
85168      ZWE    Africa  ...           61.49                   0.571
85169      ZWE    Africa  ...           61.49                   0.571
85170      ZWE    Africa  ...           61.49                   0.571

[85171 rows x 59 columns]

In [1]:
SELECT *
FROM covid_vac.csv

iso_code continent  ... life_expectancy human_development_index
0          AFG      Asia  ...           64.83                   0.511
1          AFG      Asia  ...           64.83                   0.511
2          AFG      Asia  ...           64.83                   0.511
3          AFG      Asia  ...           64.83                   0.511
4          AFG      Asia  ...           64.83                   0.511
...        ...       ...  ...             ...                     ...
85166      ZWE    Africa  ...           61.49                   0.571
85167      ZWE    Africa  ...           61.49                   0.571
85168      ZWE    Africa  ...           61.49                   0.571
85169      ZWE    Africa  ...           61.49                   0.571
85170      ZWE    Africa  ...           61.49                   0.571

[85171 rows x 37 columns]

In [2]:
SELECT *
FROM covid_deaths.csv
WHERE Continent IS NOT NULL
ORDER BY 3, 4

iso_code continent  ... life_expectancy human_development_index
0          AFG      Asia  ...           64.83                   0.511
1          AFG      Asia  ...           64.83                   0.511
2          AFG      Asia  ...           64.83                   0.511
3          AFG      Asia  ...           64.83                   0.511
4          AFG      Asia  ...           64.83                   0.511
...        ...       ...  ...             ...                     ...
81055      ZWE    Africa  ...           61.49                   0.571
81056      ZWE    Africa  ...           61.49                   0.571
81057      ZWE    Africa  ...           61.49                   0.571
81058      ZWE    Africa  ...           61.49                   0.571
81059      ZWE    Africa  ...           61.49                   0.571

[81060 rows x 59 columns]

In [3]:
SELECT location, date, total_cases, new_cases, total_deaths, population 
FROM covid_deaths.csv
WHERE Continent IS NOT NULL
ORDER BY 1, 2

location       date  total_cases  new_cases  total_deaths  population
0      Afghanistan 2020-02-24          1.0        1.0           NaN  38928341.0
1      Afghanistan 2020-02-25          1.0        0.0           NaN  38928341.0
2      Afghanistan 2020-02-26          1.0        0.0           NaN  38928341.0
3      Afghanistan 2020-02-27          1.0        0.0           NaN  38928341.0
4      Afghanistan 2020-02-28          1.0        0.0           NaN  38928341.0
...            ...        ...          ...        ...           ...         ...
81055     Zimbabwe 2021-04-26      38102.0       16.0        1560.0  14862927.0
81056     Zimbabwe 2021-04-27      38164.0       62.0        1565.0  14862927.0
81057     Zimbabwe 2021-04-28      38191.0       27.0        1565.0  14862927.0
81058     Zimbabwe 2021-04-29      38235.0       44.0        1567.0  14862927.0
81059     Zimbabwe 2021-04-30      38257.0       22.0        1567.0  14862927.0

[81060 rows x 6 columns]

## Total cases vs. total deaths

- Shows likelihood of dying if you contract COVID in your country



In [3]:
SELECT location, date, total_cases, total_deaths, 
(CAST(total_deaths AS float) / total_cases) *100  AS DeathPercentage 
FROM covid_deaths.csv
WHERE location LIKE '%States%'
ORDER BY 1, 2

location       date  total_cases  total_deaths  DeathPercentage
0    United States 2020-01-22            1           NaN              NaN
1    United States 2020-01-23            1           NaN              NaN
2    United States 2020-01-24            2           NaN              NaN
3    United States 2020-01-25            2           NaN              NaN
4    United States 2020-01-26            5           NaN              NaN
..             ...        ...          ...           ...              ...
460  United States 2021-04-26     32124869      572740.0         1.782856
461  United States 2021-04-27     32175725      573381.0         1.782030
462  United States 2021-04-28     32230850      574340.0         1.781957
463  United States 2021-04-29     32289049      575194.0         1.781390
464  United States 2021-04-30     32346971      576232.0         1.781409

[465 rows x 5 columns]

## Total cases vs. Population

- What percentage of the population has got covid in the US
- We got to 1% with 3311312 covid cases by 2020-07-12



In [5]:
SELECT location, date, population, total_cases, 
       (total_cases::FLOAT/population)*100 AS cases_population_percentage 
FROM covid_deaths.csv 
WHERE location LIKE '%States%' 
ORDER BY 1, 2

location       date  ...  total_cases  cases_population_percentage
0    United States 2020-01-22  ...            1                 3.021124e-07
1    United States 2020-01-23  ...            1                 3.021124e-07
2    United States 2020-01-24  ...            2                 6.042247e-07
3    United States 2020-01-25  ...            2                 6.042247e-07
4    United States 2020-01-26  ...            5                 1.510562e-06
..             ...        ...  ...          ...                          ...
460  United States 2021-04-26  ...     32124869                 9.705320e+00
461  United States 2021-04-27  ...     32175725                 9.720685e+00
462  United States 2021-04-28  ...     32230850                 9.737339e+00
463  United States 2021-04-29  ...     32289049                 9.754921e+00
464  United States 2021-04-30  ...     32346971                 9.772421e+00

[465 rows x 5 columns]

## Countries with the highest infection rate compared to population

- Countries with the highest infection starts with Andorra, with 17% of population infected. 

In [6]:
SELECT location, population, MAX(total_cases) AS highest_inf_count, 
       MAX((total_cases::FLOAT/population))*100 AS perc_pop_count
FROM covid_deaths.csv
-- WHERE location LIKE '%States%'
GROUP BY location, population
ORDER BY perc_pop_count DESC; 

location  population  highest_inf_count  perc_pop_count
0                Anguilla     15002.0                NaN             NaN
1                   Aruba    106766.0                NaN             NaN
2                 Bermuda     62273.0                NaN             NaN
3          Cayman Islands     65720.0                NaN             NaN
4                 Curacao    164100.0                NaN             NaN
..                    ...         ...                ...             ...
214       Solomon Islands    686878.0               20.0        0.002912
215                 Samoa    198410.0                3.0        0.001512
216               Vanuatu    307150.0                4.0        0.001302
217  Micronesia (country)    115021.0                1.0        0.000869
218              Tanzania  59734213.0              509.0        0.000852

[219 rows x 4 columns]

## Continent with the highest death count per population

In [7]:
SELECT Continent, MAX(CAST(total_deaths as int)) as total_death_count
FROM covid_deaths.csv
-- WHERE location LIKE '%States%'
WHERE Continent IS NOT NULL 
GROUP BY Continent 
ORDER BY total_death_count DESC; 

continent  total_death_count
0  North America             576232
1  South America             403781
2           Asia             211853
3         Europe             127775
4         Africa              54350
5        Oceania                910

- Let's break it down even more by location and see if North America has 576232 death count. 

In [1]:
SELECT location, MAX(CAST(total_deaths as int)) as total_death_count
FROM covid_deaths.csv
-- WHERE location LIKE '%States%'
WHERE Continent IS NOT NULL 
GROUP BY location 
ORDER BY total_death_count DESC; 

location  total_death_count
0          Anguilla                NaN
1             Aruba                NaN
2           Bermuda                NaN
3    Cayman Islands                NaN
4           Curacao                NaN
..              ...                ...
205           Timor                3.0
206            Fiji                2.0
207          Bhutan                1.0
208         Grenada                1.0
209         Vanuatu                1.0

[210 rows x 2 columns]

- We can observe that there are 576232 death count in United States, making them the number 1 country with the highest death count. 
- However, we can observe that in the continent with the highest death count query, North America is only counting USA and not Canada. 

In [8]:
SELECT location, MAX(CAST(total_deaths as int)) as total_death_count
FROM covid_deaths.csv
-- WHERE location LIKE '%States%'
WHERE Continent IS NULL 
GROUP BY location 
ORDER BY total_death_count DESC; 

location  total_death_count
0           World            3180238
1          Europe            1016750
2   North America             847942
3  European Union             688896
4   South America             672415
5            Asia             520286
6          Africa             121784
7         Oceania               1046
8   International                 15

## Global numbers

In [9]:
SELECT  date, SUM(new_cases) AS total_cases, SUM(new_deaths) AS total_deaths, 
        CAST(SUM(new_deaths) AS float) / CAST(SUM(new_cases) AS float) *100 AS death_percentage 
FROM covid_deaths.csv
WHERE Continent IS NOT NULL
GROUP BY date
ORDER BY 1, 2 

date  total_cases  total_deaths  death_percentage
0   2020-01-01          NaN           NaN               NaN
1   2020-01-02          NaN           NaN               NaN
2   2020-01-03          NaN           NaN               NaN
3   2020-01-04          NaN           NaN               NaN
4   2020-01-05          NaN           NaN               NaN
..         ...          ...           ...               ...
481 2021-04-26     682784.0       11156.0          1.633899
482 2021-04-27     844470.0       14487.0          1.715514
483 2021-04-28     905992.0       15719.0          1.735004
484 2021-04-29     897602.0       14990.0          1.670005
485 2021-04-30     879014.0       14573.0          1.657880

[486 rows x 4 columns]

- First time in the records showing total cases with covid was on 2020-01-23 with 98 cases and 1 death. 

In [10]:
SELECT  SUM(new_cases) AS total_cases, SUM(new_deaths) AS total_deaths, 
        CAST(SUM(new_deaths) AS float) / CAST(SUM(new_cases) AS float) *100 AS death_percentage 
FROM covid_deaths.csv
WHERE Continent IS NOT NULL
ORDER BY 1, 2 

total_cases  total_deaths  death_percentage
0  150574977.0     3180206.0          2.112042

- Across the world, there's 2% of death due to COVID

## Total population vs. vaccinations

- How many people across the world have been vaccinated?



In [11]:
WITH pop_vs_vacc (continent, location, date, population, new_vaccinations, people_vacc)
AS (
SELECT cd.continent, cd.location, cd.date, cd.population, cv.new_vaccinations, 
		SUM(CAST(cv.new_vaccinations AS int)) OVER(PARTITION BY cd.location ORDER BY cd.location, cd.date) AS people_vacc
FROM covid_deaths.csv cd
JOIN covid_vac.csv cv
ON cd.date = cv.date
AND cd.location = cv.location
WHERE cd.continent IS NOT NULL
ORDER BY 2, 3 )

SELECT *, (CAST(people_vacc AS float)/population)*100 AS perc_people_vacc
FROM pop_vs_vacc

continent     location  ... people_vacc  perc_people_vacc
0          Asia  Afghanistan  ...         NaN               NaN
1          Asia  Afghanistan  ...         NaN               NaN
2          Asia  Afghanistan  ...         NaN               NaN
3          Asia  Afghanistan  ...         NaN               NaN
4          Asia  Afghanistan  ...         NaN               NaN
...         ...          ...  ...         ...               ...
81055    Africa     Zimbabwe  ...    410296.0          2.760533
81056    Africa     Zimbabwe  ...    432625.0          2.910766
81057    Africa     Zimbabwe  ...    456699.0          3.072739
81058    Africa     Zimbabwe  ...    476283.0          3.204503
81059    Africa     Zimbabwe  ...    499028.0          3.357535

[81060 rows x 7 columns]

## Create table

In [2]:
CREATE TABLE perc_pop_vacc (
  continent NVARCHAR(255),
  location NVARCHAR(255),
  date DATETIME,
  population NUMERIC, 
  new_vaccinations NUMERIC,
  people_vacc NUMERIC
);

INSERT INTO perc_pop_vacc (continent, location, date, population, new_vaccinations, people_vacc)
SELECT cd.continent, cd.location, cd.date, cd.population, cv.new_vaccinations, 
       SUM(CAST(cv.new_vaccinations AS int)) OVER(PARTITION BY cd.location ORDER BY cd.location, cd.date) AS people_vacc
FROM covid_deaths.csv cd
JOIN covid_vac.csv cv
  ON cd.date = cv.date
  AND cd.location = cv.location
WHERE cd.continent IS NOT NULL
ORDER BY 2, 3;

SELECT *, (people_vacc/population)*100 AS perc_people_vacc
FROM perc_pop_vacc;



continent     location  ... people_vacc  perc_people_vacc
0          Asia  Afghanistan  ...         NaN               NaN
1          Asia  Afghanistan  ...         NaN               NaN
2          Asia  Afghanistan  ...         NaN               NaN
3          Asia  Afghanistan  ...         NaN               NaN
4          Asia  Afghanistan  ...         NaN               NaN
...         ...          ...  ...         ...               ...
81055    Africa     Zimbabwe  ...    410296.0          2.760533
81056    Africa     Zimbabwe  ...    432625.0          2.910766
81057    Africa     Zimbabwe  ...    456699.0          3.072739
81058    Africa     Zimbabwe  ...    476283.0          3.204503
81059    Africa     Zimbabwe  ...    499028.0          3.357535

[81060 rows x 7 columns]

## Create a view

In [4]:
CREATE VIEW perc_pop_vacc AS 
SELECT cd.continent, cd.location, cd.date, cd.population, cv.new_vaccinations, SUM(CAST(cv.new_vaccinations AS int)) OVER(PARTITION BY cd.location ORDER BY cd.location, cd.date) AS people_vacc
FROM covid_deaths.csv cd
JOIN covid_vac.csv cv
ON cd.date = cv.date
AND cd.location = cv.location
WHERE cd.continent IS NOT NULL;

SELECT *
FROM perc_pop_vacc

continent location       date  population  new_vaccinations  people_vacc
0        Europe  Belarus 2020-02-28   9449321.0               NaN          NaN
1        Europe  Belarus 2020-02-29   9449321.0               NaN          NaN
2        Europe  Belarus 2020-03-01   9449321.0               NaN          NaN
3        Europe  Belarus 2020-03-02   9449321.0               NaN          NaN
4        Europe  Belarus 2020-03-03   9449321.0               NaN          NaN
...         ...      ...        ...         ...               ...          ...
81055      Asia    Timor 2021-04-26   1318442.0               NaN          NaN
81056      Asia    Timor 2021-04-27   1318442.0               NaN          NaN
81057      Asia    Timor 2021-04-28   1318442.0               NaN          NaN
81058      Asia    Timor 2021-04-29   1318442.0               NaN          NaN
81059      Asia    Timor 2021-04-30   1318442.0               NaN          NaN

[81060 rows x 6 columns]